pkg/kubelet/server/server.go

```diff
@@ -729,9 +731,36 @@ func (s *Server) getContainerLogs(request *restful.Request, response *restful.Re
 		response.WriteError(http.StatusInternalServerError, fmt.Errorf("unable to convert %v into http.Flusher, cannot show logs", reflect.TypeOf(response)))
 		return
 	}
-	fw := flushwriter.Wrap(response.ResponseWriter)
+
+	var (
+		stdout io.Writer
+		stderr io.Writer
+		fw     = flushwriter.Wrap(response.ResponseWriter)
+	)
+	if utilfeature.DefaultFeatureGate.Enabled(features.SplitStdoutAndStderr) {
+		wantedStream := logOptions.Stream
```

Comment:
Also check that TailLines is not specified when the stream is not `all`

Reply:
The `logOptions` has been validated above at line 712, which already checks `TailLines` : 
https://github.com/knight42/kubernetes/blob/c78307c50b1e77f8f98161b1fc054be8f7f16a21/pkg/apis/core/v1/validation/validation.go#L151-L155

Do we need to validate again here?

Reply:
nope, that one is good